carapace-server/src/main/java/org/carapaceproxy/server/HttpProxyServer.java

```diff
@@ -119,7 +119,7 @@
     private int adminLogRetentionDays = 90;
     private boolean adminServerEnabled;
     private int adminServerHttpPort = -1;
-    private String adminServerHost = "localhost";
+    private String adminServerHost = resolveAdminServerHost();
```

Comment:
this is bad, you are performing actions with side effects/expensive operations inside the implicit initialization of the object.
You should do this while reading the configuration, and only if needed.
we have to perform this lookup only if the configured hostname is 0.0.0.0 and only in order to publish an useful value on zk metadata.
If the user configures 0.0.0.0 he is expecting to bind on 0.0.0.0